### PR TITLE
Jira8792

### PR DESCRIPTION
**Pull request by sudhirgiri2911**
Added the content update in the course batch creation.

### CODE REVIEW

File: actors/common/src/main/java/org/sunbird/learner/actors/coursebatch/CourseBatchManagementActor.java [hunk 1]
```diff
@@ -139,6 +137,30 @@ private void createCourseBatch(Request actorMessage) {
     if (courseNotificationActive()) {
       batchOperationNotifier(courseBatch, null);
     }
+    if (CourseBatchUtil.doOperationInEkStepCourse(contentDetails, courseBatch, true)
```
Comment (amolnin) [inline actors/common/src/main/java/org/sunbird/learner/actors/coursebatch/CourseBatchManagementActor.java]: Write conditions in reverse -> Constant first - followed by calculated value.

File: actors/common/src/main/java/org/sunbird/learner/util/CourseBatchUtil.java [hunk 2]
```diff
@@ -28,4 +33,40 @@ public static void syncCourseBatchForeground(String uniqueId, Map<String, Object
             + esResponse,
         LoggerEnum.INFO.name());
   }
+
+  public static String doOperationInEkStepCourse(
```
Comment (amolnin) [inline actors/common/src/main/java/org/sunbird/learner/util/CourseBatchUtil.java]: Return boolean instead -> true - means success, false - means operation did not succeed

File: actors/common/src/main/java/org/sunbird/learner/actors/coursebatch/CourseBatchManagementActor.java [hunk 3]
```diff
@@ -139,6 +137,30 @@ private void createCourseBatch(Request actorMessage) {
...
+        .equalsIgnoreCase(JsonKey.SUCCESS)) {
+      courseBatch.setCountIncrementStatus(true);
+      try {
+        courseBatch.setCountIncrementDate(new SimpleDateFormat("yyyy-MM-dd").format(new Date()));
```
Comment (amolnin) [inline actors/common/src/main/java/org/sunbird/learner/actors/coursebatch/CourseBatchManagementActor.java]: Create a constant date formatter for the whole class.

File: actors/common/src/main/java/org/sunbird/learner/actors/coursebatch/CourseBatchManagementActor.java [hunk 4]
```diff
@@ -139,6 +137,30 @@ private void createCourseBatch(Request actorMessage) {
...
+        Map<String, Object> courseBatchMap =
+            new ObjectMapper().convertValue(courseBatch, Map.class);
+        Response response = courseBatchDao.update(courseBatchMap);
+
+        if (((String) response.get(JsonKey.RESPONSE)).equalsIgnoreCase(JsonKey.SUCCESS)) {
```
Comment (amolnin) [inline actors/common/src/main/java/org/sunbird/learner/actors/coursebatch/CourseBatchManagementActor.java]: Reverse the condition

File: actors/common/src/main/java/org/sunbird/learner/actors/coursebatch/CourseBatchManagementActor.java [hunk 5]
```diff
@@ -139,6 +137,30 @@ private void createCourseBatch(Request actorMessage) {
...
+          CourseBatchUtil.syncCourseBatchForeground(
+              (String) courseBatchMap.get(JsonKey.ID), courseBatchMap);
+        } else {
+          ProjectLogger.log(
```
Comment (amolnin) [inline actors/common/src/main/java/org/sunbird/learner/actors/coursebatch/CourseBatchManagementActor.java]: Create message formatter for common portion of the Logging message

File: actors/common/src/main/java/org/sunbird/learner/actors/coursebatch/CourseBatchManagementActor.java [hunk 6]
```diff
@@ -139,6 +137,30 @@ private void createCourseBatch(Request actorMessage) {
...
+              "CourseBatchManagementActor:createCourseBatch: Course batch not synced to ES as response is not successful",
+              LoggerEnum.INFO.name());
+        }
+      } catch (Exception ex) {
```
Comment (amolnin) [inline actors/common/src/main/java/org/sunbird/learner/actors/coursebatch/CourseBatchManagementActor.java]: Catch specific exceptions -> can use pipe-seperated syntax - but catch exceptions explicitly

File: actors/common/src/main/java/org/sunbird/learner/util/CourseBatchUtil.java [hunk 7]
```diff
@@ -28,4 +33,40 @@ public static void syncCourseBatchForeground(String uniqueId, Map<String, Object
...
+      Map<String, Object> ekStepContent, CourseBatch courseBatch, boolean increment) {
+    String name = ProjectUtil.getConfigValue(JsonKey.SUNBIRD_INSTALLATION);
+    String contentName = "";
+    String response = "";
+    Map<String, String> ekstepHeader = CourseBatchSchedulerUtil.headerMap;
+    if (courseBatch.getEnrollmentType().equals(ProjectUtil.EnrolmentType.open.getVal())) {
```
Comment (amolnin) [inline actors/common/src/main/java/org/sunbird/learner/util/CourseBatchUtil.java]: write condition in reverse

File: actors/common/src/main/java/org/sunbird/learner/util/CourseBatchUtil.java [hunk 8]
```diff
@@ -28,4 +33,40 @@ public static void syncCourseBatchForeground(String uniqueId, Map<String, Object
...
+    }
+    try {
+      ProjectLogger.log("updating content details to Ekstep start", LoggerEnum.INFO.name());
+      String contentUpdateBaseUrl = ProjectUtil.getConfigValue(JsonKey.EKSTEP_BASE_URL);
+      response =
+          HttpUtil.sendPatchRequest(
+              contentUpdateBaseUrl
+                  + PropertiesCache.getInstance().getProperty(JsonKey.EKSTEP_CONTENT_UPDATE_URL)
+                  + courseBatch.getCourseId(),
+              "{\"request\": {\"content\": {\"" + contentName + "\": " + val + "}}}",
```
Comment (amolnin) [inline actors/common/src/main/java/org/sunbird/learner/util/CourseBatchUtil.java]: Create map, use ObjectMapper writeValueAsString to convert to json

File: actors/common/src/main/java/org/sunbird/learner/util/CourseBatchUtil.java [hunk 9]
```diff
@@ -28,4 +33,40 @@ public static void syncCourseBatchForeground(String uniqueId, Map<String, Object
...
+      contentName = "c_" + name + "_private_batch_count";
+    }
+    int val = (int) ekStepContent.getOrDefault(contentName, 0);
+    if (increment) {
+      val = val + 1;
+    } else {
+      if (val != 0) val = val - 1;
+    }
+    try {
+      ProjectLogger.log("updating content details to Ekstep start", LoggerEnum.INFO.name());
```
Comment (amolnin) [inline actors/common/src/main/java/org/sunbird/learner/util/CourseBatchUtil.java]: use standard logging prefix with message formaatter

File: actors/common/src/main/java/org/sunbird/learner/actors/coursebatch/CourseBatchManagementActor.java [hunk 10]
```diff
@@ -139,6 +137,30 @@ private void createCourseBatch(Request actorMessage) {
...
+        ProjectLogger.log(
+            "CourseBatchManagementActor:createCourseBatch: Exception occurred with error message = "
+                + ex.getMessage(),
+            ex);
+      }
+    }
```
Comment (amolnin) [inline actors/common/src/main/java/org/sunbird/learner/actors/coursebatch/CourseBatchManagementActor.java]: break-down the function into smaller chunks - function has grown to too many lines.

File: actors/common/src/main/java/org/sunbird/learner/util/CourseBatchUtil.java (hunk 2 above)
Comment (amolnin) [inline actors/common/src/main/java/org/sunbird/learner/util/CourseBatchUtil.java]: CourseBatchSchedulerUtil.doOperationInEkStepCourse -> how is this function different then mentioned here?

File: actors/common/src/main/java/org/sunbird/learner/actors/coursebatch/CourseBatchManagementActor.java [hunk 11]
```diff
@@ -854,4 +851,32 @@ private void validateContentOrg(List<String> createdFor) {
       }
     }
   }
+
+  @SuppressWarnings("unchecked")
+  private void syncWithEkstepAndUpdateDB(CourseBatch courseBatch) {
```
Comment (amolnin) [inline actors/common/src/main/java/org/sunbird/learner/actors/coursebatch/CourseBatchManagementActor.java]: syncToEkStepAndUpdateDB

File: actors/common/src/main/java/org/sunbird/learner/actors/coursebatch/CourseBatchManagementActor.java [hunk 12]
```diff
@@ -854,4 +851,32 @@ private void validateContentOrg(List<String> createdFor) {
...
+
+        if (JsonKey.SUCCESS.equalsIgnoreCase((String) response.get(JsonKey.RESPONSE))) {
+          CourseBatchUtil.syncCourseBatchForeground(
+              (String) courseBatchMap.get(JsonKey.ID), courseBatchMap);
+        } else {
+          ProjectLogger.log(
+              "CourseBatchManagementActor:createCourseBatch: Course batch not synced to ES as response is not successful",
+              LoggerEnum.INFO.name());
+        }
+      } catch (Exception ex) {
```
Comment (amolnin) [inline actors/common/src/main/java/org/sunbird/learner/actors/coursebatch/CourseBatchManagementActor.java]: catch multi-exception  in single catch block 
- but do not catch generic exception

File: actors/common/src/main/java/org/sunbird/learner/actors/coursebatch/CourseBatchManagementActor.java [hunk 13]
```diff
@@ -854,4 +851,32 @@ private void validateContentOrg(List<String> createdFor) {
...
+        ProjectLogger.log(
+            "CourseBatchManagementActor:createCourseBatch: Exception occurred with error message = "
```
Comment (amolnin) [inline actors/common/src/main/java/org/sunbird/learner/actors/coursebatch/CourseBatchManagementActor.java]: create message formatter for prefix: CourseBatchManagementActor:createCourseBatch:

File: actors/common/src/main/java/org/sunbird/learner/util/CourseBatchSchedulerUtil.java [hunk 14]
```diff
@@ -220,7 +219,7 @@ public static String doOperationInEkStepCourse(
       ProjectLogger.log(
           "EKstep content not found for course id==" + courseId, LoggerEnum.INFO.name());
     }
-    return response;
+    return response.equalsIgnoreCase(JsonKey.SUCCESS);
```
Comment (amolnin) [inline actors/common/src/main/java/org/sunbird/learner/util/CourseBatchSchedulerUtil.java]: JsonKey.SUCCESS.equalsIgnoreCase(response)

File: actors/common/src/main/java/org/sunbird/learner/actors/coursebatch/CourseBatchManagementActor.java [hunk 15]
```diff
@@ -139,6 +138,7 @@ private void createCourseBatch(Request actorMessage) {
     if (courseNotificationActive()) {
       batchOperationNotifier(courseBatch, null);
     }
+    syncToEkstepAndUpdateDB(courseBatch);
```
Comment (bvinayakumar) [inline actors/common/src/main/java/org/sunbird/learner/actors/coursebatch/CourseBatchManagementActor.java]: Rename method to updateBatchCount

File: actors/common/src/main/java/org/sunbird/learner/actors/coursebatch/CourseBatchManagementActor.java [hunk 16]
```diff
@@ -433,9 +433,8 @@ private void getCourseBatchDetails(Request actorMessage) {
 
   private int setCourseBatchStatus(String startDate) {
     try {
-      SimpleDateFormat format = new SimpleDateFormat("yyyy-MM-dd");
-      Date todaydate = format.parse(format.format(new Date()));
-      Date requestedStartDate = format.parse(startDate);
+      Date todaydate = DATE_FORMAT.parse(DATE_FORMAT.format(new Date()));
```
Comment (bvinayakumar) [inline actors/common/src/main/java/org/sunbird/learner/actors/coursebatch/CourseBatchManagementActor.java]: todayDate

**Pull request merged**